modules/llama_cpp_plugin/include/state.hpp

```diff
@@ -0,0 +1,24 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef LLAMA_CPP_PLUGIN_HPP
+#define LLAMA_CPP_PLUGIN_HPP
+
+#include "openvino/runtime/ivariable_state.hpp"
+#include "compiled_model.hpp"
+
+namespace ov {
+    namespace llama_cpp_plugin {
+        class LlamaCppState : public IVariableState {
+        public:
+            LlamaCppState() = delete;
+            LlamaCppState(const std::shared_ptr<const LlamaCppModel>& model_ptr) : m_model_ptr(model_ptr), IVariableState("llama_cpp_state") {}
+            void reset() override {
```

Comment:
we could also implement get_state and set_state

Reply:
I suppose that this could be done to manually set the kv-cache, but I would need a reference to a real use case first so that I could set up some kind of acceptance testing for this.